sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/ByteKeyRangeTracker.java

```diff
@@ -64,23 +87,36 @@ public synchronized ByteKeyRange checkpoint() {
   /**
    * Attempts to claim the given key.
    *
-   * <p>Must be larger than the last successfully claimed key.
+   * <p>Must be larger than the last attempted key. Note that passing in {@link ByteKey#EMPTY}
```

Comment:
Why do we care about whether the key is greater than last attempted rather than last claimed?

Would you mind adding a short javadoc defining what `lastAttemptedKey` represents? I believe it represents the last attempted claim, regardless of whether it succeeded or not.

Reply:
Yes, will add the javadoc in the follow-up PR since tests passed.